Tools/TestWebKitAPI/Tests/WebKitCocoa/FindInPage.mm

```diff
@@ -334,7 +334,14 @@ static NSUInteger replaceMatches(WKWebView *webView, NSArray<FindMatch> *matches
 #if HAVE(UIFINDINTERACTION)
 
 // FIXME: (rdar://95125552) Remove conformance to _UITextSearching.
-@interface WKWebView () <UITextSearching>
+@interface WKWebView () <UITextSearching, _UITextSearching>
+@property (nonatomic, readonly) _UIFindInteraction *_findInteraction;
+@property (nonatomic, readwrite, setter=_setFindInteractionEnabled:) BOOL _findInteractionEnabled;
+@property (nonatomic, readonly) CALayer *_layerForFindOverlay;
+
+- (void)_addLayerForFindOverlay;
+- (void)_removeLayerForFindOverlay;
+- (void)_requestRectForFoundTextRange:(UITextRange *)ranges completionHandler:(void (^)(CGRect))completionHandler;
```

Comment:
Can we just include `WKWebViewPrivate.h` to pull these in? (slightly surprised that isn't already the case)

Reply:
We do already include WKWebViewPrivate.h. The challenge is that it only declares these when the private SDK is available:
```
#if __has_include(<UIKit/_UIFindInteraction.h>)
@property (nonatomic, readonly) _UIFindInteraction *_findInteraction WK_API_AVAILABLE(ios(16.0));
@property (nonatomic, readwrite, setter=_setFindInteractionEnabled:) BOOL _findInteractionEnabled WK_API_AVAILABLE(ios(16.0));

@property (nonatomic, readonly) CALayer *_layerForFindOverlay WK_API_AVAILABLE(ios(16.0));

- (void)_requestRectForFoundTextRange:(UITextRange *)ranges completionHandler:(void (^)(CGRect))completionHandler WK_API_AVAILABLE(ios(16.0));

- (void)_addLayerForFindOverlay WK_API_AVAILABLE(ios(16.0));
- (void)_removeLayerForFindOverlay WK_API_AVAILABLE(ios(16.0));
#endif
```

So we can't rely solely on WKWebViewPrivate.h without emitting a bunch of redeclarations in our private headers.

I spent some time trying to move usage over to `UIFindInteraction` (rdar://95125552), which is new API in iOS 16. I was able to clean things up nicely and avoid these redeclarations, but that broke the open source iOS 15 build. I could spend a bit more time trying to get that working, but it might be better to wait until next year when we drop support for building with the iOS 15 SDK.

Reply:
I see!

We should just be able to change the `__has_include` to `TARGET_OS_IOS && __IPHONE_OS_VERSION_MIN_REQUIRED >= 160000`.

Reply:
Our style checker doesn't like `__IPHONE_OS_VERSION_MIN_REQUIRED` in private headers:

    ERROR: Source/WebKit/UIProcess/API/Cocoa/WKWebViewPrivate.h:500:  Misplaced OS version check. Please use a named macro in one of headers in the wtf/Platform.h suite of files or an appropriate internal file.  [build/version_check] [5]

I spent some more time messing with UIFindInteraction, and was able to clean up our remaining usage of `_UITextSearching` and `_UIFindInteraction` SPI in https://github.com/WebKit/WebKit/pull/3494/commits/98a38143d27572483303206082f81850cfe70954. That, plus switching to checking for the public header (`__has_include(<UIKit/UIFindInteraction.h>)`) seems to be working well.

Reply:
I think we can ignore that style warning. We can't use the platform macros in this file, so I don't think the warning applies here.

We cannot remove the SPI since we have clients that still rely on it.

Reply:
Got it! PR updated with your feedback, here's a diff: https://github.com/WebKit/WebKit/compare/9c5d061133b9eb82a7593d8e7b605a377a37562c..7e2ce29be192a73eedba1d402920529b0a5fb649